src/shared/pkcs11-util.c

```diff
@@ -508,22 +508,65 @@ int pkcs11_token_read_x509_certificate(
 }
 #endif
 
+static int pkcs11_token_find_private_key_from_template(
+        CK_FUNCTION_LIST *m,
+        CK_SESSION_HANDLE session,
+        CK_ATTRIBUTE_PTR template,
+        CK_ULONG templateCount,
+        CK_OBJECT_HANDLE_PTR objects,
+        CK_ULONG objectCount,
+        CK_ULONG *ret_nobjects) {
```

Comment:
as per our usual coding style, please indent fuction parameter lists to double of the usual indentation, i.e. 16 spaces